public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -102,6 +102,17 @@ export function transformV2(
     (df) => isHeatmapResult(df, request)
   );
 
+  // this works around the fact that we only get back frame.name with le buckets when legendFormat == {{le}}...which is not the default
+  heatmapResults.forEach((df) => {
+    if (df.name == null) {
+      let le = df.fields.find((f) => f.name === 'Value')?.labels?.le;
```

Comment:
should this loop through each field in the frame and set the field name to the value of `le` if that label exists? setting the frame name seems like it wouldn't work with wide frames

Reply:
scratch that. i see what's going on now in `sortSeriesByLabel`. it looks like this will be updated if/when the plugin backend outputs wide frames

Reply:
that would require more changes since sorting has always relied on frame names.

https://github.com/grafana/grafana/blob/bdfa127ee5c11948568d83579e60351792e65f4f/public/app/plugins/datasource/prometheus/result_transformer.ts#L637-L638

and we actually rename the Value fields based on frame names later:

https://github.com/grafana/grafana/blob/bdfa127ee5c11948568d83579e60351792e65f4f/public/app/plugins/datasource/prometheus/result_transformer.ts#L593